rootfs-builder/README.md

```diff
@@ -124,3 +85,88 @@ must be met:
    $ # build image based rootfs created above
    $ ./image-builder/image_builder.sh "${PWD}/fedora_rootfs"
    ```
+
+## Adding support for a new guest OS
+
+The `rootfs.sh` script will check for immediate sub-directories
+containing the following expected files:
+
+- A `bash(1)` script called `rootfs_lib.sh`
+
+  This file must contain a function called `build_rootfs()` which
+  must receive as its first argument the path to where the rootfs will be
+  created.
+
+  Path: `rootfs-builder/<distro>/rootfs_lib.sh`.
+
+
+- A `bash(1)` script called `config.sh`
+
+  This represents the specific configuration for `<distro>`. It must
+  provide configuration specific variables for the user to modify as needed.
+  The `config.sh` file will be loaded before executing `build_rootfs()` to
+  provide all the needed configuration to the function.
+
+  Path: `rootfs-builder/<distro>/config.sh`.
+
+### Create template files
+
+To create a directory with the expected file structure run:
+
+```
+$ make -f template/Makefile  ROOTFS_BASE_NAME=my_new_awesome_rootfs
+```
+
+After running the command above, a new directory will be created in
+`rootfs-builder/my_new_awesome_rootfs/`. To verify it is one of the
+options to build a rootfs run `./rootfs.sh -h`, it will show
+`my_new_awesome` as one of the options to use it for:
+
+```
+$ sudo ./rootfs.sh <distro>
+```
+
+### Modify template files
+
+After the new directory structure has been created:
+
+- If needed, add configuration variables to
+  `rootfs-builder/my_new_awesome_rootfs/config.sh`.
+
+- Implement the stub `build_rootfs()` function from
+  `rootfs-builder/my_new_awesome_rootfs/rootfs_lib.sh`.
+
+### Expected rootfs directory content
+
+After the function `build_rootfs` is called, the script expects the
+rootfs directory to contain `/sbin/init` and `/sbin/kata-agent` binaries.
+
+### (optional) Customise the rootfs
+
+For particular uses cases developers may want to modify the guest OS.
+
+#### Adding extra packages
+
+To add additional packages, use one of these methods:
+
+- Use the environment variable `EXTRA_PKGS` to provide a list of space separated
+  packages to be installed.
+
+  *Note: The package names may vary among Linux distributions, the extra
+  package names must exist in the base OS flavor you use to build the
+  rootfs* from.
+
+  Example:
+
+  ```
+  $ EXTRA_PKGS="vim emacs" ./rootfs-builder/rootfs.sh -r ${PWD}/myrootfs fedora
```

Comment:
I wonder if we should remove Emacs from the example? :)

Reply:
What you mean because "why would you need emacs if you have vim?"

/me runs... :smile: